lib/elixir/test/elixir/string_test.exs

```diff
@@ -33,8 +33,12 @@ defmodule StringTest do
     assert String.split("foo bar ") == ["foo", "bar"]
     assert String.split(" foo bar ") == ["foo", "bar"]
     assert String.split("foo\t\n\v\f\r\sbar\n") == ["foo", "bar"]
-    assert String.split("foo" <> <<31>> <> "bar") == ["foo", "bar"]
     assert String.split("foo" <> <<194, 133>> <> "bar") == ["foo", "bar"]
+    # information separators are not considered whitespace
+    assert String.split("foo\u001Fbar") == ["foo\u001Fbar"]
+    # no-break space is excluded
```

Comment:
"no-break" –> "non-breaking"?

Reply:
Oof. It's this sort of thing which makes us app developers have to specifically add U+00A0 to the list of things to break on.

Reply:
@trejkaz as you probably know, this is actually more complex because there are cases where you don't want to split on U+00A0. For example, as an example provided by Unicode itself, it can be used to separate numbers: `$3 567,00`.

The truth is that this `split` implementation is arbitrary and it will only work on the simplest of cases. For anything that actually requires understanding the text semantics, such as line/word boundary breaking, then you need to follow the rules in both Unicode and CLDR.

Reply:
As @josevalim says, U+00A0 is specifically intended as a non-breaking space. As are U+202F, U+2007 and U+2060.

Reply:
But nonetheless a space, and thus we end up wanting to break on it anyway, if doing real text processing. But yeah, it is more complex than that too. Actually in the case of numbers you can have a breaking space there too, and people expect it to be processed as a number. If you have two words separated by a non-breaking space, they also expect it to be processed the same as a space. *shrug*

Reply:
That's exactly the point. If you are doing real text processing, then you surely don't want to use `String.split`, regardless if it includes breaking or non-breaking spaces. Adding no-break spaces won't make it any more suitable for processing text. Not even passing your own list of characters would make it any better.